bika/lims/browser/bika_listing.py

```diff
@@ -1321,6 +1321,13 @@ def _folderitems(self, full_objects=False):
             results_dict['state_title'] = st_title
 
             results_dict['class'] = {}
+
+            # As far as I am concerned, adapters for IFieldIcons are only used
+            # for Analysis content types. Since AnalysesView is not using this
+            # "classic" folderitems from bikalisting anymore, this logic has
+            # been added in AnalysesView. Even though, this logic hasn't been
+            # removed from here, cause this _folderitems function is marked as
+            # deprecated, so it will be eventually removed alltogether.
             for name, adapter in getAdapters((obj,), IFieldIcons):
                 auid = obj.UID() if hasattr(obj, 'UID') and callable(
                     obj.UID) else None
```

Comment:
why don't you use `api.get_uid` here and for which objects does this return `None`?

Reply:
This is old code I haven't touched because is stale (see the comment I added)

Reply:
Ah, sorry, I see...